api/api.go

```diff
@@ -71,6 +71,9 @@ func (a *APIServer) RegisterAPIHandler(handler APIHandler) error {
 			Method: "GET",
 			Path:   "/api/" + name,
 			HandlerFunc: func(w http.ResponseWriter, r *auth.AuthenticatedRequest) {
+				if r.TLS != nil {
```

Comment:
moving this if statement into a http wrapper in the http package will ease to maintain the tls code. Probably wrapping the router.